x-pack/plugin/profiler/src/main/java/org/elasticsearch/xpack/profiler/GetProfilingResponse.java

```diff
@@ -166,6 +178,11 @@ public Exception getError() {
 
     @Override
     public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params params) {
+        // Calculate the sample rate.
+        double samplingRate = 0.0;
+        if (samplingExponent != 0) {
+            samplingRate = Math.pow(0.2, samplingExponent);
```

Comment:
Instead of doing this here, can't we just pass the result of `EventsIndex.getSampleRate()` to `GetProfilingResponseBuilder` and use that value?